server/test/unit/common/test_app_config.py

```diff
@@ -133,3 +134,23 @@ def test_get_config_vars_from_aws_secrets(self, mock_get_secret_key):
             self.assertEqual(config.server_config.app__flask_secret_key, "mock_flask_secret")
             self.assertEqual(config.server_config.authentication__params_oauth__client_secret, "mock_oauth_secret")
             self.assertEqual(config.default_dataset_config.user_annotations__hosted_tiledb_array__db_uri, "mock_db_uri")
+
+    def test_backend_base_url(self):
+
+        # test the backend_base_url feature, and that it can contain a path
+        c = AppConfig()
```

Comment:
For readability, please avoid using single character variable names.

Reply:
> Friendly ping about this question:
> 
> > Curious, why are APIs served from separate dataroots? The API doesn't change per dataroot no?

I think it's mostly historical reasons for that.
It could be different, something like  e.g. /api/v0.2/schema?dataset=datasetname.cxg.
Maybe now that we are separating out the frontend and backend it makes sense to revisit this.

Reply:
yeah, the URL is super bizarre and non intuitive. we should be aiming for something like:
`https://api.cellxgene.dev.single-cell.czi.technology/cellxgene/userinfo` or
`https://api.cellxgene.dev.single-cell.czi.technology/cellxgene/schema?dataset=datasetname.cxg`. Or even a dataset UUID.

There are lots of way to handle versioning but one clean way is for the versions to go into the Accept and Content-Type headers.